kura/org.eclipse.kura.linux.position/src/main/java/org/eclipse/kura/linux/position/PositionServiceImpl.java

```diff
@@ -318,12 +329,21 @@ else if(UsbDeviceRemovedEvent.USB_EVENT_DEVICE_REMOVED_TOPIC.contains(event.getT
 				
 				s_logger.debug("ModemGpsEnabledEvent");
 				
-				m_properties.put("port", event.getProperty(ModemGpsEnabledEvent.Port));
-				m_properties.put("baudRate", event.getProperty(ModemGpsEnabledEvent.BaudRate));
-				m_properties.put("bitsPerWord", event.getProperty(ModemGpsEnabledEvent.DataBits));
-				m_properties.put("stopBits", event.getProperty(ModemGpsEnabledEvent.StopBits));
-				m_properties.put("parity", event.getProperty(ModemGpsEnabledEvent.Parity));
-				m_properties.put("modem", "true");
+				m_properties.put(PORT, event.getProperty(ModemGpsEnabledEvent.Port));
+				m_properties.put(BAUDRATE, event.getProperty(ModemGpsEnabledEvent.BaudRate));
+				m_properties.put(BITSPERWORD, event.getProperty(ModemGpsEnabledEvent.DataBits));
+				m_properties.put(STOPBITS, event.getProperty(ModemGpsEnabledEvent.StopBits));
+				m_properties.put(PARITY, event.getProperty(ModemGpsEnabledEvent.Parity));
+				m_properties.put(MODEM, "true");
+				
+				if (m_gpsDevice != null) {
+					Properties currentConfigProps = m_gpsDevice.getConnectConfig();
+					Properties serialProperties = getSerialConnectionProperties(m_properties);
+					if ((currentConfigProps != null) && (serialProperties != null)) {
+						checkProperties(currentConfigProps, serialProperties);
```

Comment:
@pierantoniomerlino , the checkProperties() is a void method that only compares current properties with serial properties and logs the "same configuration, no need to reconfigure GPS device" if properties are equal. Don't we want to call the updated() method (in the handleEvents()) based on result of this comparison?